File: RedSheet.DbEntities/Models/ModuleContent.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ModuleContents", Schema = "dbo")]
	[Serializable]
	public partial class ModuleContent  
	{

	public string Abkhazian { get; set; } 
 
	[Required] 
	[MaxLength(10)]
	public string Action { get; set; } 

	public string Afar { get; set; } 

	public string Afrikaans { get; set; } 

	public string Amharic { get; set; } 

	public string English { get; set; } 

	public string French { get; set; } 
 
	[MaxLength(20)]
	public string LanguageContentType { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ModuleContentId { get; set; } 

	public Nullable<int> ServerMessageId { get; set; } 

	public string Zulu { get; set; } 
 
	[RelationshipTableAttribue("ApplicationModules","dbo")]
	//Relationships
	public int ApplicationModuleId { get; set; } 
 
	[ForeignKey("ApplicationModuleId")]
	public ApplicationModule ApplicationModule { get; set; } 
 
	[RelationshipTableAttribue("LanguageContents","dbo")]
	//Relationships
	public int LanguageContentId { get; set; } 
 
	[ForeignKey("LanguageContentId")]
	public LanguageContent LanguageContent { get; set; } 

	public ModuleContent()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/module-content.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ApplicationModule, LanguageContent,  } from './'
export class ModuleContent {
    constructor(moduleContent?: ModuleContent )  {
        let properties = [ "abkhazian", "action", "afar", "afrikaans", "amharic", "english", "french", "languageContentType", "moduleContentId", "serverMessageId", "zulu", "applicationModuleId", "languageContentId",];
        for (let property of properties)
            if (moduleContent && moduleContent[property])
                this[property] = moduleContent[property];
    }
 
	abkhazian : string =   undefined;
 
    @required()
    @maxLength(10)
	action : string =   undefined;
 
	afar : string =   undefined;
 
	afrikaans : string =   undefined;
 
	amharic : string =   undefined;
 
	english : string =   undefined;
 
	french : string =   undefined;
 
    @maxLength(20)
	languageContentType : string =   undefined;
 
	moduleContentId : number =   0 ;
 
	serverMessageId : number =   undefined;
 
	zulu : string =   undefined;
 
    @range(0,2147483647)
	applicationModuleId : number =   undefined;
	applicationModule : ApplicationModule  ;
 
    @range(0,2147483647)
	languageContentId : number =   undefined;
	languageContent : LanguageContent  ;


}
